pavelib/quality.py

```diff
@@ -903,117 +851,28 @@ def _lint_output(linter, count, violations_list, is_html=False, limit=0):
             title = f"Quality Report: {linter}\n"
             violations_str = ''.join(violations_list)
             violations_count_str = "Violations: {count}\n"
-            fail_line = f"FAILURE: {linter} count should be {limit}\n"
+            fail_line = f"FAILURE: {linter} count should be 0\n"
 
         violations_count_str = violations_count_str.format(count=count)
 
         lines.extend([sep, title, sep, violations_str, sep, violations_count_str])
 
-        if count > limit > -1:
+        if count > 0:
             lines.append(fail_line)
         lines.append(sep + '\n')
         if is_html:
             lines.append('</body>')
 
         return ''.join(lines)
 
-    # If pylint reports exist, use those
-    # Otherwise, `diff-quality` will call pylint itself
     (count, violations_list) = _get_pylint_violations(clean=False)
-    _, upper_violations_limit, _, _ = _parse_pylint_options(options)
 
     # Print total number of violations to log
-    print(_lint_output('pylint', count, violations_list, limit=upper_violations_limit))
-    if count > upper_violations_limit > -1:
-        diff_quality_pass = False
-        failure_reasons.append('Too many total violations.')
-
-    # ----- Set up for diff-quality pylint call -----
-    # Set the string to be used for the diff-quality --compare-branch switch.
-    compare_branch = getattr(options, 'compare_branch', 'origin/master')
-    compare_commit = sh(f'git merge-base HEAD {compare_branch}', capture=True).strip()
-    if sh('git rev-parse HEAD', capture=True).strip() != compare_commit:
-        compare_branch_string = f'--compare-branch={compare_commit}'
-
-        # Set the string, if needed, to be used for the diff-quality --fail-under switch.
-        diff_threshold = int(getattr(options, 'percentage', -1))
-        percentage_string = ''
-        if diff_threshold > -1:
-            percentage_string = f'--fail-under={diff_threshold}'
-
-        pylint_files = get_violations_reports("pylint")
-        pylint_reports = ' '.join(pylint_files)
-        if not run_diff_quality(
-            violations_type="pylint",
-            reports=pylint_reports,
-            percentage_string=percentage_string,
-            branch_string=compare_branch_string,
-            dquality_dir=dquality_dir
-        ):
-            diff_quality_pass = False
-            failure_reasons.append('Pylint violation(s) were found in the lines of code that were added or changed.')
-
-        eslint_files = get_violations_reports("eslint")
```

Comment:
These reports are also being generated in the respective functions. Diff-quality used to generate these for the comparison.

Reply:
Hmm, `run_eslint` will fail the job if the count is over the threshold, but not if there were failures in the lines changed.  @davidjoy , is that "fail quality if there are eslint issues on lines you added or touched" behavior something we want to preserve?  The quality checks would be simpler and faster if we omit that, maybe we should just update the threshold to be closer to the current actual warning count instead?  (I see 4,854 warnings on master and a current threshold of 5,300.)  But that would increase the risk of serious new issues slipping through in PRs that happen to also fix older existing issues.

Reply:
Hmm, it's hard to know what to say.  In a world without amnesty, failing linting if there are failures on the lines you added or touched feels like the _right_ condition, in that it's equivalent to saying "there are errors now, and there weren't before."

If that's prohibitively slow in the monolith for whatever reason because we need to account for amnesty by only running eslint against the changes in the PR, then changes in the overall error count seems like a reasonable proxy (though, as you point out, it's hard to keep the threshold reasonable).  If it were possible to tell whether it went "up" rather than if it met some threshold, that'd probably be more ideal.

(In the end I trust your judgement)